jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/DefaultControllerEnricher.java

```diff
@@ -145,4 +146,10 @@ public void create(PlatformMode platformMode, KubernetesListBuilder builder) {
     return spec.getSpec().getContainers().stream().map(Container::getName).collect(Collectors.toList());
   }
 
+  private static String getDefaultContainerRestartPolicy(String controllerType) {
+    if (controllerType != null && controllerType.equalsIgnoreCase(ControllerType.JOB.name())) {
+      return DEFAULT_JOB_RESTART_POLICY;
+    }
+    return null;
+  }
```

Comment:
this should be delegated to the handler (it makes no sense having an if-else condition here, given that we have specific handlers to take care of specifics).

Reply:
JobHandler seems to be generating pod template spec using PodTemplateHandler:
https://github.com/eclipse/jkube/blob/b4c31fe43d6ee853bcf85128eb31727176a9d258/jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/handler/JobHandler.java#L67-L70

We don't have any way of knowing whether PodTemplate is generated for Job or any other kind of controller in PodTemplateHandler. That's why I'm setting restart policy from DefaultControllerEnricher from where handlers are invoked.